### DISCUSSION TITLE
[Detector Question]: which tensorrt model to use?

### DISCUSSION THREAD

**Discussion by vajonam** [blakeblackshear/frigate#7867] [Ask A Question]
### Describe the problem you are having

Just curious, with the previous release we had access to yolov4, and now have yolov7. I noticed the the yolov7-320 model had some more false positives. More than the previous yolov4-416 model I was using. Moving to yolov7-640 helped with the false positives, but the inference time on my GPU is up to 70ms from <20ms. My question is, if should I keep with the yolov4-416 model or move over to the yolov7 mode, what are others expereinces here?

**Comment by vajonam** [comment]
@NateMeyer  320 seems a bit too small for me and my camera resolutions, too many false positives and 640 might be overkill interns of GPU horse power I have. 416 in yolov4  was a good compromise for me, any chance we can get that added back as an option for yolov7? 

I have added this locally for my container environment, overriding 

```
- /opt/docker/configs/frigate/override/download_yolo.sh:/usr/local/src/tensorrt_demos/yolo/download_yolo.sh
```

added 

```
if [[ -z "$TARGET_MODEL" ]] || [[ $TARGET_MODEL =~ .*"yolov7-416"*. ]]; then
    echo "Creating yolov7-416.cfg and yolov7-416.weights"
    cat yolov7.cfg | sed -e '6s/batch=64/batch=1/' | sed -e '8s/width=640/width=416/' | sed -e '9s/height=640/height=416/' > yolov7-416.cfg
    ln -sf yolov7.weights yolov7-416.weights
fi
```

I will report back on this and how it works for possible inclusion into your repo?

**Reply by NateMeyer** [reply]
Let me know how it goes.  If that works well, we can add that to tensorrt_demos repo.

**Reply by vajonam** [reply]
So far its been working well for me. However its detection some "squirrels" as person, but that is easily filtered out using min size for a person, but other than that quite reliable, its about 1/2 the inference time, but doesn't seem like half the precision. I dont have a quantitative test setup using existing videos. but that is where I am at with this, would vote to add it.

**Reply by vajonam** [reply]
the false positives I was getting with shadows and IR flight flashes as persons are now gone. So I would say its helped in that regard without increase inference time too much.

**Reply by Dinth** [reply]
Hiya. It looks like 416 is already in the download_yolo.sh in the main repository. Maybe it can be added to the documentation too?

**Comment by twister36** [comment]
I agree yolov7-320 seems to have allot of false positives. I too am going to try the other models...

I see in the online documentation a list of the models available, and yolov4-416 is available (@vajonam). However do we have a better list that defines each models accuracy vs speed?

**Reply by vajonam** [reply]
I have not quantified this, but created a yolov7-416 model that I have been using with minimal false positives.  Instructions [above](https://github.com/blakeblackshear/frigate/discussions/7867#discussioncomment-7038159).

**Reply by twister36** [reply]
I'll have to give it a try. I did find a few links that describe what the different models do and what they are used for. Right now I'm testing yolo7-640z.

**Reply by Sceptersax** [reply]
@twister36 how's yolov7-640? Significantly more accurate or just slightly?

**Reply by Dinth** [reply]
yolov7-320 works perfectly fine on P640 with interference speed of 30ms, but both -416 and -640 just segfault
